File: easy/insertionsort/go/insert.go
package main

import (
	"fmt"
)

func InsertSort(nums []int) []int {
	for i := range len(nums) {
		j := i - 1
		for j >= 0 && nums[j+1] < nums[j] {
			temp := nums[j+1]
			nums[j+1] = nums[j]
			nums[j] = temp
			j--
		}
	}

	return nums
}

func main() {
	fmt.Println("This should sort the array [2, 3, 4, 1, 6]")
	fmt.Println("The output should be [1, 2, 3, 4, 6]")

	arr := []int{2, 3, 4, 1, 6}

	arr = InsertSort(arr)

	fmt.Printf("%v\n", arr)
}

File: easy/insertionsort/insert.py
class Solution:
    def __init__(self):
        self._nums: list[int] = [2, 3, 4, 1, 6, 5]

    def get_nums(self):
        return self._nums

    def sort(self) -> list[int]:
        sort_nums = self.get_nums()
        for i in range(1, len(sort_nums)):
            j = i - 1
            while j >= 0 and sort_nums[j + 1] < sort_nums[j]:
                temp = sort_nums[j + 1]
                sort_nums[j + 1] = sort_nums[j]
                sort_nums[j] = temp
                j -= 1
        return sort_nums


def main():
    insert_solution = Solution()

    result = insert_solution.sort()

    print("This is insertion sort and we are using the list[int] of [2, 3, 4, 1, 6, 5]")
    print("\nThis should return [1, 2, 3, 4, 5, 6]")
    print(f"{result}")


if __name__ == "__main__":
    main()
